hopsworks-api/src/main/java/io/hops/hopsworks/api/dataset/DatasetBuilder.java

```diff
@@ -0,0 +1,409 @@
+        case TYPE:
+          return order(a.getDatasetType().name(), b.getDatasetType().name(), sortBy.getParam());
+        case SEARCHABLE:
+          return order(a.isSearchable(), b.isSearchable(), sortBy.getParam());
+        case MODIFICATION_TIME:
+          return order(a.getAttributes().getModificationTime(), b.getAttributes().getModificationTime(),
+            sortBy.getParam());
+        case ACCESS_TIME:
+          return order(a.getAttributes().getAccessTime(), b.getAttributes().getAccessTime(), sortBy.getParam());
+        default:
+          throw new UnsupportedOperationException("Sort By " + sortBy + " not supported");
+      }
+    }
+    
+    private int order(String a, String b, AbstractFacade.OrderBy orderBy) {
+      switch (orderBy) {
+        case ASC:
+          return String.CASE_INSENSITIVE_ORDER.compare(a, b);
+        case DESC:
+          return String.CASE_INSENSITIVE_ORDER.compare(b, a);
+        default:
+          throw new UnsupportedOperationException("Order By " + orderBy + " not supported");
+      }
+    }
+    
+    private int order(Boolean a, Boolean b, AbstractFacade.OrderBy orderBy) {
+      switch (orderBy) {
+        case ASC:
+          return a.compareTo(b);
+        case DESC:
+          return b.compareTo(a);
+        default:
+          throw new UnsupportedOperationException("Order By " + orderBy + " not supported");
+      }
+    }
+    
+    private int order(Integer a, Integer b, AbstractFacade.OrderBy orderBy) {
+      switch (orderBy) {
+        case ASC:
+          return a.compareTo(b);
+        case DESC:
+          return b.compareTo(a);
+        default:
+          throw new UnsupportedOperationException("Order By " + orderBy + " not supported");
+      }
+    }
+    
+    private int order(Long a, Long b, AbstractFacade.OrderBy orderBy) {
+      switch (orderBy) {
+        case ASC:
+          return a.compareTo(b);
+        case DESC:
+          return b.compareTo(a);
+        default:
+          throw new UnsupportedOperationException("Order By " + orderBy + " not supported");
+      }
+    }
+    
+    private int order(Date a, Date b, AbstractFacade.OrderBy orderBy) {
+      switch (orderBy) {
+        case ASC:
+          return a.compareTo(b);
+        case DESC:
+          return b.compareTo(a);
+        default:
+          throw new UnsupportedOperationException("Order By " + orderBy + " not supported");
+      }
+    }
+    
+    @Override
+    public int compare(DatasetDTO a, DatasetDTO b) {
+      Iterator<DatasetFacade.SortBy> sort = sortBy.iterator();
+      int c = compare(a, b, sort.next());
+      for (; sort.hasNext() && c == 0; ) {
+        c = compare(a, b, sort.next());
+      }
+      return c;
+    }
+  }
+  
+  private DatasetDTO mergeAndApplyOffsetAndLimit(DatasetDTO dto, ResourceRequest resourceRequest, DatasetDTO dto1,
+    DatasetDTO dto2) {
+    if ((dto1 == null || dto1.getItems() == null || dto1.getItems().isEmpty()) &&
```

Comment:
Count can be > 0 even when the list is empty if limit is set

Reply:
so `count` is not the number of items in the `items` field?
actually, you are setting `count` to the number of items after limit so, in dataset case, it will always apply (https://github.com/ErmiasG/hopsworks/blob/458dc77eff7858d98f6282a4cf935c4d8a453d1d/hopsworks-api/src/main/java/io/hops/hopsworks/api/dataset/DatasetBuilder.java#L407) but I gt your point. maybe it doesn't apply for other cases

Reply:
`count` is the number of all instances of this resource after the request filters are applied. For example, if the request asks to filter all datasets with name like `abc`, and that returns 10 items then `count` will be 10. But if on the same request we say offset=5 limit=5, then 5 items will be returned but count will be 10. @ErmiasG correct me if I'm wrong :)